certora/specs/core/RewardsCoordinator.spec

```diff
@@ -0,0 +1,113 @@
+
+using ERC20 as token1;
+using ERC20PresetFixedSupply as token2;
+
+methods {
+    function _.balanceOf(address) external => DISPATCHER(true);
+    function _.transfer(address, uint256) external => DISPATCHER(true);
+    function _.transferFrom(address, address, uint256) external => DISPATCHER(true);
+
+    function _.createAVSRewardsSubmission(IRewardsCoordinator.RewardsSubmission[]) external => DISPATCHER(true);
+    function _.createRewardsForAllSubmission(IRewardsCoordinator.RewardsSubmission[]) external => DISPATCHER(true);
+    function _.submitRoot(bytes32 root, uint32 rewardsCalculationEndTimestamp) external => DISPATCHER(true);
+    function _.processClaim(IRewardsCoordinator.RewardsMerkleClaim claim, address recipient) external => DISPATCHER(true);
+
+    function _.verifyInclusionKeccak(bytes memory, bytes32, bytes32, uint256) internal => CONSTANT;
+    function cumulativeClaimed(address, address) external returns(uint256) envfree;
+
+    function _.strategyIsWhitelistedForDeposit(address) external => NONDET;
+    function _.isPauser(address) external => NONDET;
+    function _.unpauser() external => NONDET;
+    function _._ external => DISPATCH [
+        _.transferFrom(address, address, uint256),
+        _.transfer(address, uint256)
+    ] default NONDET;
+}
+
+
+// Given a processClaim passes, second processClaim reverts
+// Assumptions
+// - checkClaim(claim) == true and does not revert
+// - claim has non-empty tokenLeaves
+/// status: pass
+rule claimWithduplicateTokenLeafs(env e, IRewardsCoordinator.RewardsMerkleClaim claim, address recipient) {
+    require claim.tokenLeaves.length == 1;
+    
+    checkClaim(e, claim);
```

Comment:
what's the point of this call?

Reply:
Its to ensure that the claim is valid.
since checkClaim doesn't have @withrevert any successful paths of this rule will include checkClaim not reverting. Probably more clear by just having `require checkClaim`

Reply:
makes sense, but adding 'require' syntax would definitely be clearer